### PR TITLE
Move systemDownHandler to CorfuRuntimeParameters.

### PR DESCRIPTION
**Pull request by zalokhan**
## Overview

Description:
Moving the systemDownHandler to CorfuParameters. Making it volatile and adding additional logging.

Why should this be merged: Adding logging for debuggability.

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/runtime/view/AbstractView.java [hunk 1]
```diff
@@ -159,8 +159,12 @@ private Layout getLayoutUninterruptibly() {
 
             // Invoking the systemDownHandler if the client cannot connect to the server.
             if (--systemDownTriggerCounter <= 0) {
-                runtime.systemDownHandler.run();
+                log.info("layoutHelper: Invoking the systemDownHandler.");
+                runtime.getParameters().getSystemDownHandler().run();
             }
+            // Minimum counter value should be 0.
+            systemDownTriggerCounter = Math.max(0, systemDownTriggerCounter);
```
Comment (medhavidhawan) [inline runtime/src/main/java/org/corfudb/runtime/view/AbstractView.java]: if we are going to do it this way, then why not do this way
if(Math.max(0, --systemDownTriggerCounter) == 0) {

}

Comment (zalokhan) [reply runtime/src/main/java/org/corfudb/runtime/view/AbstractView.java]: We will not update the counter with the value then. This causes us to log negative numbers in the log statements.

Comment (medhavidhawan) [reply runtime/src/main/java/org/corfudb/runtime/view/AbstractView.java]: yes or you can do a better job by updating the number of attempts you have made so far.

Comment (zalokhan) [reply runtime/src/main/java/org/corfudb/runtime/view/AbstractView.java]: Done.

**Pull request merged**